src/transformers/models/propainter/modeling_propainter.py

```diff
@@ -0,0 +1,4513 @@
+# limitations under the License.
+"""PyTorch ProPainter model."""
+
+import itertools
+import math
+from collections import namedtuple
+from functools import reduce
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+import torch.utils.checkpoint
+import torchvision
+from torch import nn
+from torch.nn import L1Loss
+from torch.nn.functional import normalize
+from torch.nn.modules.utils import _pair, _single
+from torchvision import models as tv
+
+from ...modeling_outputs import (
+    BaseModelOutput,
+    MaskedImageModelingOutput,
+)
+from ...modeling_utils import TORCH_INIT_FUNCTIONS, PreTrainedModel
+from ...utils import (
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    logging,
+)
+from .configuration_propainter import ProPainterConfig
+
+
+logger = logging.get_logger(__name__)
+
+# General docstring
+_CONFIG_FOR_DOC = "ProPainterConfig"
+
+# Base docstring
+_CHECKPOINT_FOR_DOC = "ruffy369/propainter"
+_EXPECTED_OUTPUT_SHAPE = ["batch_size", 80, 240, 432, 3]
+
+
+# Adapted from original code at https://github.com/sczhou/ProPainter
+
+
+class ProPainterResidualBlock(nn.Module):
+    def __init__(self, config, in_channels, channels, norm_fn="group", stride=1):
+        super().__init__()
+
+        self.config = config
+
+        self.conv1 = nn.Conv2d(
+            in_channels, channels, kernel_size=3, padding=1, stride=stride
+        )
+        self.conv2 = nn.Conv2d(channels, channels, kernel_size=3, padding=1)
+        self.relu = nn.ReLU(inplace=True)
+
+        num_groups = channels // 8
+
+        if norm_fn == "group":
+            self.norm1 = nn.GroupNorm(num_groups=num_groups, num_channels=channels)
+            self.norm2 = nn.GroupNorm(num_groups=num_groups, num_channels=channels)
+            if not stride == 1:
+                self.norm3 = nn.GroupNorm(num_groups=num_groups, num_channels=channels)
+
+        elif norm_fn == "batch":
+            self.norm1 = nn.BatchNorm2d(channels)
+            self.norm2 = nn.BatchNorm2d(channels)
+            if not stride == 1:
+                self.norm3 = nn.BatchNorm2d(channels)
+
+        elif norm_fn == "instance":
+            self.norm1 = nn.InstanceNorm2d(channels)
+            self.norm2 = nn.InstanceNorm2d(channels)
+            if not stride == 1:
+                self.norm3 = nn.InstanceNorm2d(channels)
+
+        elif norm_fn == "none":
+            self.norm1 = nn.Sequential()
+            self.norm2 = nn.Sequential()
+            if not stride == 1:
+                self.norm3 = nn.Sequential()
+
+        if stride == 1:
+            self.downsample = None
+
+        else:
+            self.downsample = nn.Sequential(
+                nn.Conv2d(in_channels, channels, kernel_size=1, stride=stride),
+                self.norm3,
+            )
+
+    def forward(self, hidden_states):
```

Comment:
(extreme nit) you can use `hidden_state` of a single tensor, and `hidden_states` for a list of tensors

Reply:
done

Reply:
Oh! thank you @NielsRogge for mentioning, this would be a nice change for clarity as well. Done it in the latest commits